CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RenamedChildrenCompatibilityFile.swift

```diff
@@ -0,0 +1,84 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+let renamedChildrenCompatibilityFile = try! SourceFileSyntax(leadingTrivia: copyrightHeader) {
+  for layoutNode in SYNTAX_NODES.compactMap(\.layoutNode).filter({ $0.children.hasDeprecatedChild }) {
+    try ExtensionDeclSyntax("extension \(raw: layoutNode.type.syntaxBaseName)") {
+      for child in layoutNode.children {
+        if let deprecatedVarName = child.deprecatedVarName {
+          let childType: TypeSyntax = child.kind.isNodeChoicesEmpty ? child.syntaxNodeKind.syntaxType : "\(raw: child.name)"
+          let type = child.isOptional ? TypeSyntax("\(raw: childType)?") : TypeSyntax("\(raw: childType)")
+
+          DeclSyntax(
+            """
+            @available(*, deprecated, renamed: "\(raw: child.varName)")
+            public var \(raw: deprecatedVarName): \(raw: type) {
+              get {
+                return \(raw: child.varName)
+              }
+              set(value) {
+                \(raw: child.varName) = value
+              }
```

Comment:
Nit: This can be slightly shorter by using the implicit `newValue` variable.

```suggestion
              set {
                \(raw: child.varName) = newValue
              }
```